### PR TITLE
Nuopc merge to master

### PR DESCRIPTION
**Pull request by jedwards4b**
Merge latest nuopc-cmeps development into cime master

Test suite: scripts_regression_tests.py 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes 
User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by rljacob** [comment]
Can you summarize the changes outside of drivers/nuopc ?

File: src/share/timing/perf_mod.F90 [hunk 1]
```diff
@@ -14,6 +14,14 @@ module perf_mod
 !-----------------------------------------------------------------------
 !- Uses ----------------------------------------------------------------
 !-----------------------------------------------------------------------
+#ifdef NUOPC_INTERFACE
+#define TIMERSTART call ESMF_TraceRegionEnter
+#define TIMERSTOP  call ESMF_TraceRegionExit
+  use ESMF, only: ESMF_TraceRegionEnter, ESMF_TraceRegionExit
```
Comment (jedwards4b) [inline src/share/timing/perf_mod.F90]: replace calls to gptl with calls to ESMF timing routines.

**Comment by jgfouca** [comment]
@jedwards4b , is this ready for merge?

**Comment by jedwards4b** [comment]
@jgfouca I had a codacy failure that I wanted to fix, waiting for new results.

**Comment by jgfouca** [comment]
OK, I will merge once it checks-out.

File: scripts/lib/CIME/hist_utils.py [hunk 2]
```diff
@@ -280,8 +281,8 @@ def _compare_hists(case, from_dir1, from_dir2, suffix1="", suffix2="", outfile_s
                         logger.warning("Could not copy {} to {}".format(cprnc_log_file, casedir))
 
                 all_success = False
-
-    if num_compared == 0:
+    # PFS test may not have any history files to compare.
```
[the remark this answers is not in the document]
Comment (billsacks) [reply scripts/lib/CIME/hist_utils.py]: Does this also fix the GENERATE failure documented in #2917 ?

**Comment by jedwards4b** [comment]
@billsacks no it does not, I will submit another PR to do that.
